docs/usage/shoot_updates.md

```diff
@@ -0,0 +1,56 @@
+# Shoot Updates and Upgrades
+
+This document describes what happens during shoot updates (changes incorporated in a newly deployed Gardener version) and during shoot upgrades (changes for version controllable by end-users).
+
+## Updates
+
+Updates to all aspects of the shoot cluster happen when the gardenlet reconciles the `Shoot` resource.
+The period, i.e., how often the reconciliation is executed, depends on the configuration of the Gardener administrators/operators.
+In some Gardener installations the operators might enable "reconciliation in maintenance time window only" ([more information](shoot_maintenance.md#cluster-reconciliation)) which will result in at least one reconciliation during the time configured in the `Shoot`'s `.spec.maintenance.timeWindow` field.
+
+As end-users can only control the `Shoot` resource's specification but not the used Gardener version, they don't have any influence on which of the updates are being rolled out (other than those settings configurable in the `Shoot`).
+A Gardener operator can deploy a new Gardener version at any point in time.
+Any subsequent reconciliation of `Shoot`s will update them by rolling out the changes incorporated in this new Gardener version.
+
+Some examples for such shoot updates are:
+
+* Add a new/remove an old component to/from the shoot's control plane running in the seed, or to/from the shoot's system components running on the worker nodes.
+* Change the configuration of an existing control plane/system component.
+* Add a new/remove an old condition to/from the `.status.conditions[]` list maintained in the `Shoot` resource.
+* Restart of existing control plane/system components (this might result in a short unavailability of the Kubernetes API server (e.g., when etcd or a kube-apiserver itself is being restart)
+
+### Behavioural Changes
+
+Generally, some of such updates (e.g., configuration changes) could theoretically result in different behaviour of controllers.
+If such changes would be backwards-incompatible then we usually follow one of those approaches (depends on the concrete change):
+
+* Only apply the change for new clusters.
+* Expose a new field in the `Shoot` resource that lets users control this changed behaviour to enable it at a convenient point in time.
+* Put the change behind an alpha feature gate (disabled by default) in the gardenlet (only controllable by Gardener operators) which will be promoted to beta (enabled by default) in subsequent releases (in this case, end-users have no influence on when the behaviour changes - Gardener operators should inform their end-users and provide clear timelines when they will enable the feature gate).
+
+## Upgrades
+
+We consider shoot upgrades to change either the
+
+* Kubernetes version (`.spec.kubernetes.version`)
```

Comment:
I think the main motivation for https://github.com/gardener/gardener/issues/2827 was to specify in details the process of k8s version upgrade (machines are not rolled for k8s patch version upgrade and machines are rolled for k8s minor version upgrade). Or does this info exist in another place?

Reply:
I think it would be very helpful for Shoot owners to add documentation that explains which configuration changes on the Shoot (k8s version + changes in the worker config) causes either an in place update or a rolling update.
Rolling Update
 - OS  & k8s minor version  -> 
-  Worker data volumes (type,size,encryption - what about the name?) -> 
- Worker volume

 In place
-  Worker containerRuntimes (OCI)  added / removed
- Worker caBundle
- patch k8s version(not 100 sure here)
 
... you name it.

This is also provider specific for <providerConfig> so that would require a doc in each provider extension too.

Reply:
Yes, thanks, good points, I missed that completely. Added it now, PTAL :)